### ISSUE TITLE
Associate! Symbiota proposed modifications

### ISSUE THREAD

**Issue by seltmann** [ParasiteTracker/other-projects-modeling-biotic-interactions#4]
We proposed some database changes to Symbiota as part of the Assocate! Project.  These are documented here.

**Comment by seltmann** [comment]
These are the documented fields from the Associate! proposal for updating Symbiota to handle biotic associations. 

associd | primary key (PK)
-- | --
occid (required) | FK for base occurrence
occidassociate (optional) | FK for associated occurrence; used only when association is an occurrence managed within same Symbiota instance; associated occurrences managed within external systems (e.g. iDigBio, GBIF) can be represented using link in resourceURL and dynamicProperties fields.
relationship | controlled vocabulary will be used
basisOfRecord | controlled vocabulary defining the specific nature of the associate occurrence, which will often be different than the base occurrence (e.g. HumanObservation, LiteratureCitation, etc)
identifier | globally unique identifier (GUID) representing the association. Will typically be a UUID automatically generated by Symbiota, though can be over ridden when GUID was generated outside of the Symbiota environment.
resourceurl | URL of relationships that were established outside of the Symbiota environment
verbatimsciname | verbatim name which may be full scientific name, abbreviated name, common name, or other non-standard format. Name is indexed to taxonomic authority post import
locationOnHost | controlled vocabulary, linked out to ontologies
condition | controlled vocabulary, linked out to ontologies
dateEmerged | date/time field recording date and time of emergence
dynamicProperties | JSON string consisting key-value pairs representing any dynamic property of the association. This can be used to store extended DwC field data for the associated occurrence or custom properties that are specific to a certain association (e.g. gall number, color, etc).
notes | general notes

**Comment by seltmann** [comment]
@neilcobb working with GloBI and SCAN the past few weeks, there are a few simple modifications to Symbiota that are worth noting. First dynamicProperties are not represented in the public view of a specimen record. Need to check that dynamicProperties, associatedTaxa are both in the public view.

Second, agree that we need an associatedOccurence field that is editable.

**Comment by neilcobb** [comment]
Evin will be at the meeting next week and this can be on his list to do.

**Comment by seltmann** [comment]
A few other additions that would make life helpful:
associatedTaxa and dynamicProperties fields need to be bigger, dynamicProperties open without having to click the +

**Comment by seltmann** [comment]
Did not mean to close, reopening.

**Comment by seltmann** [comment]
In the automatic lookup for associatedTaxa, change the comma as a separator to semi-colon.

**Comment by seltmann** [comment]
dynamicProperties included in table edit view

**Comment by seltmann** [comment]
<img width="964" alt="Screen Shot 2019-09-27 at 7 07 59 AM" src="https://user-images.githubusercontent.com/1044474/65775731-96048a00-e0f5-11e9-9181-7b28717245c1.png">

associatedOccurences is a URI, not just a catalog number. Do we need a separate field? Is this a link out in Symbiota?

**Comment by seltmann** [comment]
Link out to GloBI record

**Comment by seltmann** [comment]
Link out to DOI

**Comment by neilcobb** [comment]
this is not used in sin by Yoda as far as I know. So when it is implemented I assumed it would be implemented the same way that it is in gbif idigbio arctos etcetera

Get Outlook for Android<https://aka.ms/ghei36>




On Fri, Sep 27, 2019 at 7:09 AM -0700, "Katja Seltmann" <notifications@github.com<mailto:notifications@github.com>> wrote:


Link out to DOI

**Comment by seltmann** [comment]
<img width="967" alt="Screen Shot 2019-09-27 at 7 27 50 AM" src="https://user-images.githubusercontent.com/1044474/65777194-65721f80-e0f8-11e9-882e-35c89862726f.png">

**Comment by neilcobb** [comment]
yes, so we now have two fields to add and two to show on front end

**Comment by evindunn** [comment]
@seltmann  @neilcobb I've added 'Associated Occurrences' and 'Associated Taxa' to the individual & table views in SCAN. I'll work on actual data entry now.

**Comment by seltmann** [comment]
Thats amazing @evindunn. Please let me know if you want me to test anything. I have a few TODOs on my list including making sure the GloBI known relationships are online.

**Comment by seltmann** [comment]
@neilcobb @evindunn I created a more detailed document of what I see is a path forward for implementing robust interactions in Symbiota. This is based on looking at how Arctos has implemented interactions and understanding we have limited resources for a module.

https://docs.google.com/document/d/1S9FQdHiOb3lsH7yCoC-wvirV_72tbNCDMZPjDx2qh-c/edit?usp=sharing

**Comment by neilcobb** [comment]
this looks very complete

**Comment by seltmann** [comment]
@evindunn let me know if you have more to show. I can meet any day.
